### PR TITLE
[Fairseq] Add fairseq TTS model to hub

### PR DESCRIPTION
**Pull request by Narsil**

### CODE REVIEW

File: api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py [hunk 1]
```diff
@@ -11,7 +12,9 @@ class TextToSpeechPipeline(Pipeline):
     def __init__(self, model_id: str):
         # hardcoded stuff can later be moved to a config
         model, cfg, task = load_model_ensemble_and_task_from_hf_hub(
-            model_id, arg_overrides={"vocoder": "griffin_lim", "fp16": False}
+            model_id,
+            arg_overrides={"vocoder": "griffin_lim", "fp16": False},
+            cache_dir=os.getenv("HUGGINGFACE_HUB_CACHE"),
```
Comment (Narsil) [inline api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py:17]: Use `HUGGINGFACE_HUB` to make docker it the cache (See the dockerfile this is defined as `/data` and this is mounted as an external disk on the kube cluster to make the model files persistent.

File: api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py [hunk 2]
```diff
@@ -39,6 +42,8 @@ def __call__(self, inputs: str) -> Tuple[np.array, int]:
         """
         tokenized_inputs = self._tokenize(inputs)
 
+        if not tokenized_inputs:
+            return np.zeros((0,)), self.sampling_rate
```
Comment (Narsil) [inline api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py:46]: If the inputs is `''` then there's a crash later. We can simply return nil in that case

**Review by patrickvonplaten — COMMENTED** [review]
Thanks a lot!

File: api-inference-community/docker_images/fairseq/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
-api-inference-community==0.0.19
-g2p_en
-git+git://github.com/pytorch/fairseq.git
+api-inference-community==0.0.20
+g2p_en==2.1.0
+librosa==0.8.1
```
Comment (Narsil) [inline api-inference-community/docker_images/fairseq/requirements.txt:3]: Pinning all version + adding librosa which wasn't there.
+ use latest `api-inference-community`.

File: api-inference-community/tests/test_dockers.py [hunk 4]
```diff
@@ -79,6 +79,12 @@ def test_espnet(self):
             "espnet/kamo-naoyuki_mini_an4_asr_train_raw_bpe_valid.acc.best",
         )
 
+    def test_fairseq(self):
```
Comment (Narsil) [inline api-inference-community/tests/test_dockers.py:82]: Adding a docker test (Checking that launching the docker works and most use cases show correct errors + cache is working.

File: api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py (hunk 1 above)
Comment (patrickvonplaten) [reply api-inference-community/docker_images/fairseq/app/pipelines/text_to_speech.py:17]: Thanks for the explanation

**Pull request merged**